jprotoc/README.md

```diff
@@ -130,4 +130,43 @@ Using the Jdk8 Protoc generator
 3. Reference the java 8 client stubs like this:
     ```java
     MyServiceGrpc8.GreeterCompletableFutureStub stub = MyServiceGrpc8.newCompletableFutureStub(channel);
-    ```
\ No newline at end of file
+    ```
+    
+Implementing Custom Protoc Options
+==================================
+1. Create a custom option as per https://developers.google.com/protocol-buffers/docs/reference/java-generated#extension.
+For example:
+
+    ```
+    syntax = "proto3";
+    
+    package com.example.proto.options;
+    
+    import "google/protobuf/descriptor.proto";
+    
+    option java_multiple_files = true;
+    option java_outer_classname = "ServerOptionsProto";
+    option java_package = "com.example.proto.options";
+    
+    extend google.protobuf.FileOptions {
+        ServerOptions server = 50621;
+    }
+    
+    message ServerOptions {
+        // Java classname
+        string name = 1;
+    }
+    ```
+
```

Comment:
Do I have to run the protobuf generator as part of the build process for my protobuf plugin? Do I need to add this to my plugin project's POM?

Reply:
I'm not familiar with Maven but I don't see any reason anything in the build would have to change. All the processing is done at `protoc` run-time when `protoc` calls the `protoc` plugin.

Reply:
Maybe I'm not understanding correctly. Where does the type `com.example.proto.options.ServerOptionsProto.server` come from in the example below? Perhaps you could add an elaborated example in `jprotoc-test`?

Reply:
Oh, that's generated by `protoc`. Those not creating custom options don't need to run `protoc` to generate the related Java code for the option. I'm assuming those creating custom options would know how to run `protoc` to do that. I can add a blurb about running `protoc` if you want.

Reply:
I see...My JProtoc Protobuf generator project would itself contain Protobuf generated code (it's proto all the way down!). A line to say "Run the Protoc generator on the Extension proto" should suffice.

Reply:
Done.